src/idl_gen_general.cpp

```diff
@@ -828,6 +828,33 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
     code += ") + _bb.";
     code += lang_.get_bb_position;
     code += ", _bb)); }\n";
+
+    // recreate both methods for the prefixed size version
+    std::string ps_method_name = FunctionStart('G') + "etSizePrefixedRootAs" +
+                                 struct_def.name;
+    std::string ps_method_signature = "  public static " + struct_def.name + " " +
+                                      ps_method_name;
+
+    // create convenience method that doesn't require an existing object
+    code += ps_method_signature + "(ByteBuffer _psbb) ";
+    code += "{ return " + ps_method_name + "(_psbb, new " + struct_def.name+ "()); }\n";
+
+    // use a slice that skips the size, then proceed as normal
+    code += ps_method_signature + "(ByteBuffer _psbb, " + struct_def.name + " obj) { ";
+    code += "ByteBuffer _bb = _psbb." + FunctionStart('S') + "lice(); ";
+    if (lang_.language == IDLOptions::kCSharp) {
+      code += "_bb.Position = 4; ";
```

Comment:
constant?

Reply:
I have added a constant for Java and C# each. The Java one had to be explicitly `public` though, because the other ones are package-private.